internal/inspect/policyset.go

```diff
@@ -0,0 +1,57 @@
+// Copyright 2021-2024 Zenauth Ltd.
+// SPDX-License-Identifier: Apache-2.0
+
+package inspect
+
+import (
+	"fmt"
+	"sort"
+
+	responsev1 "github.com/cerbos/cerbos/api/genpb/cerbos/response/v1"
+	runtimev1 "github.com/cerbos/cerbos/api/genpb/cerbos/runtime/v1"
+	"github.com/cerbos/cerbos/internal/namer"
+	"github.com/cerbos/cerbos/internal/policy"
+)
+
+func PolicySets() *PolicySet {
+	return &PolicySet{
+		results: make(map[string]*responsev1.InspectPoliciesResponse_Result),
+	}
+}
+
+type PolicySet struct {
+	results map[string]*responsev1.InspectPoliciesResponse_Result
+}
+
+// Inspect inspects the given policy set and caches the inspection related information internally.
+func (ps *PolicySet) Inspect(pset *runtimev1.RunnablePolicySet) error {
+	if pset == nil {
+		return fmt.Errorf("policy set is nil")
+	}
+
+	actions := policy.ListPolicySetActions(pset)
+	if len(actions) > 0 {
+		sort.Strings(actions)
+	}
+
+	variables := policy.ListPolicySetVariables(pset)
+	if len(variables) > 0 {
+		sort.Slice(variables, func(i, j int) bool {
+			return sort.StringsAreSorted([]string{variables[i].Name, variables[j].Name})
+		})
+	}
+
+	if len(actions) > 0 || len(variables) > 0 {
+		ps.results[namer.PolicyKeyFromFQN(pset.Fqn)] = &responsev1.InspectPoliciesResponse_Result{
+			Actions:   actions,
+			Variables: variables,
+		}
+	}
```

Comment:
If the user asked to inspect policy `foo`, it should exist in the final result regardless of whether it has any actions or variables.

```suggestion

		ps.results[namer.PolicyKeyFromFQN(pset.Fqn)] = &responsev1.InspectPoliciesResponse_Result{
			Actions:   actions,
			Variables: variables,
		}

```